service/matching/matching_engine.go

```diff
@@ -852,11 +830,12 @@ func (e *matchingEngineImpl) UpdateWorkerVersioningRules(
 	if req.GetTaskQueue() != request.GetTaskQueue() {
 		return nil, serviceerror.NewInternal("Task Queue does not match Task Queue in wrapped command")
 	}
-	taskQueue, err := newTaskQueueID(ns.ID(), req.GetTaskQueue(), enumspb.TASK_QUEUE_TYPE_WORKFLOW)
+
+	taskQueueFamily, err := tqid.NewTaskQueueFamily(req.GetNamespace(), req.GetTaskQueue())
 	if err != nil {
 		return nil, err
 	}
-	tqMgr, err := e.getTaskQueuePartitionManager(ctx, taskQueue, normalStickyInfo, true)
+	tqMgr, err := e.getTaskQueuePartitionManager(ctx, taskQueueFamily.TaskQueue(enumspb.TASK_QUEUE_TYPE_WORKFLOW).RootPartition(), true)
```

Comment:
it looks weird to be just assuming workflow+root here.. I see it's correct because of how we route this in client, but I think a comment would be nice to assure readers